pkg/playground/react-demo-file-upload.tsx

```diff
@@ -0,0 +1,140 @@
+
+            setFiles(oldFiles => {
+                return {
+                    [file.name]: { file, progress: 0, cancel: () => abort.abort() },
+                    ...oldFiles,
+                };
+            });
+
+            try {
+                await upload(destination, file.stream(), (progress) => {
+                    if (performance.now() < next_progress)
+                        return;
+                    next_progress = performance.now() + 200; // only rerender every 200ms
+                    setFiles(oldFiles => {
+                        const oldFile = oldFiles[file.name];
+                        return {
+                            ...oldFiles,
+                            [file.name]: { ...oldFile, progress },
+                        };
+                    });
+                }, abort.signal);
+            } catch (exc) {
+                cockpit.assert(exc instanceof Error);
+                if (exc instanceof DOMException && exc.name == 'AbortError') {
+                    setAlert({ variant: "warning", title: 'Aborted', message: '' });
+                } else {
+                    setAlert({ variant: "danger", title: 'Upload Error', message: exc.message });
+                }
+            } finally {
+                setFiles(oldFiles => {
+                    const copy = { ...oldFiles };
+                    delete copy[file.name];
+                    return copy;
+                });
+            }
+        }));
+
+        // Reset input field in the case a download was cancelled and has to be re-uploaded
+        // https://stackoverflow.com/questions/26634616/filereader-upload-same-file-again-not-working
+        event.target.value = "";
+    };
+
+    return (
+        <>
+            <Flex direction={{ default: "column" }}>
+                <FileAutoComplete className="upload-file-dest" value={dest} onChange={setDest} />
+                <Button
+                  id="upload-file-btn"
+                  variant="secondary"
+                  icon={<UploadIcon />}
+                  isDisabled={Object.keys(files).length !== 0}
+                  isLoading={Object.keys(files).length !== 0}
+                  onClick={handleClick}
+                >
+                    {_("Upload")}
+                </Button>
+                <input
+                  ref={ref} type="file"
+                  hidden multiple onChange={onUpload}
+                />
+            </Flex>
+            {alert !== null &&
+            <Alert variant={alert.variant}
+                   title={alert.title}
+                   timeout={3000}
+                   actionClose={<AlertActionCloseButton onClose={() => setAlert(null)} />}
+            >
+                <p>{alert.message}</p>
+            </Alert>
+            }
+            {Object.keys(files).map((key, index) => {
+                const file = files[key];
+                return (
+                    <React.Fragment key={index}>
```

Comment:
I thought ESLint wanted us to write fragments as `<>` ... `</>` now, but apparently we still use this syntax in a few places.

Reply:
This comment is definitely not for me :)

Reply:
If you pass a `key` you can't use `<>` this is valid code

Reply:
Aha! Thanks!